web-app/packages/lib/src/modules/user/store.ts

```diff
@@ -380,167 +381,86 @@ export const useUserStore = defineStore('userModule', {
+      if (this.loggedUser?.id === undefined) return
 
-        const value = JSON.stringify({
-          ...oldValue,
-          [this.loggedUser?.username]: payload.id
-        })
-        const expires = new Date()
-        // cookies expire in one year
-        expires.setFullYear(expires.getFullYear() + 1)
-        cookies.set(COOKIES_CURRENT_WORKSPACE, value, {
-          expires
+      const lastSeenWorkspace = [...this.lastWorkspaces.value]
+        .sort((a, b) => {
+          return b.lastSeen - a.lastSeen
         })
-      }
+        .find((item) => item.userId === this.loggedUser?.id)
+
+      return (
+        lastSeenWorkspace && this.getWorkspaceById({ id: lastSeenWorkspace.id })
+      )
     },
 
-    async setFirstWorkspace() {
-      if (this.workspaces.length > 0) {
-        await this.setWorkspace({
-          workspaceId: this.workspaces[0].id
-        })
-      }
+    async cleanupLastSeenWorkspaces() {
+      this.lastWorkspaces.value = this.lastWorkspaces.value.filter(
+        (item) => item.userId !== this.loggedUser?.id
+      )
     },
 
+    /**
+     * Checks the current workspace that the user is in.
+     *
+     * Tries to get the current workspace ID from the cookies.
+     * If not found, tries to use the preferred workspace ID from the user profile.
+     * If still not found, defaults to using the first workspace in the user's list of workspaces.
+     *
+     * Sets the current workspace in the store, and saves to cookies if needed.
+     */
     async checkCurrentWorkspace() {
       try {
         await this.getWorkspaces()
-        const currentWorkspaceFromCookie =
-          await this.getUserWorkspaceFromCookies()
+        let foundWorkspace = this.getLastWorkspaceFromStorage()
         if (this.workspaces.length > 0) {
-          let foundWorkspace
           let skipSavingInCookies = true
-          if (currentWorkspaceFromCookie !== undefined) {
-            // try to use workspace stored in cookies
-            foundWorkspace = this.getWorkspaceById({
-              id: parseInt(currentWorkspaceFromCookie)
-            })
-          }
           if (!foundWorkspace) {
             // try to use preferred_workspace from user profile response
-            const preferredWorkspaceId = await this.getPreferredWorkspaceId()
-            if (preferredWorkspaceId) {
-              foundWorkspace = this.getWorkspaceById({
-                id: preferredWorkspaceId
-              })
-              if (foundWorkspace?.id) {
-                // do not skip saving in cookies when using preferredWorkspace
-                skipSavingInCookies = false
-              }
+            const preferredWorkspaceId = this.loggedUser?.preferred_workspace
+            foundWorkspace = this.getWorkspaceById({
+              id: preferredWorkspaceId
+            })
+            if (foundWorkspace?.id) {
+              // do not skip saving in cookies when using preferredWorkspace
+              skipSavingInCookies = false
```

Comment:
do we need this?

Reply:
I can rename variables . some workspace changes might not be persisted in storage due to specific logic.

Reply:
cool, just rename in that case